File: src/main/java/io/github/takusan23/clickmanaita/block/ClickManaitaBaseBlock.java
package io.github.takusan23.clickmanaita.block;

import net.minecraft.core.BlockPos;
import net.minecraft.world.InteractionHand;
import net.minecraft.world.InteractionResult;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.level.Level;
import net.minecraft.world.level.block.Block;
import net.minecraft.world.level.block.state.BlockBehaviour;
import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.world.phys.BlockHitResult;
import org.jetbrains.annotations.NotNull;

/**
 * クリックまな板ブロック
 * <p>
 * コンストラクタは引数が二個ある方を利用してください
 */
public class ClickManaitaBaseBlock extends Block {

    /**
     * ドロップ数
     */
    private int dropSize = 2;

    /**
     * 引数が二個ある方を利用してね
     */
    @Deprecated
    public ClickManaitaBaseBlock(BlockBehaviour.Properties properties) {
        super(properties);
    }

    /**
     * @param properties  クリエイティブタブの設定など
     * @param argDropSize どろっぷする個数
     */
    public ClickManaitaBaseBlock(BlockBehaviour.Properties properties, int argDropSize) {
        super(properties);
        dropSize = argDropSize;
    }

    /**
     * 右クリックしたとき。まな板の本命機能
     * <p>
     * なんか非推奨になってるんだけど何？
     */
    @Override
    public @NotNull InteractionResult use(BlockState p_57450_, Level p_57451_, BlockPos p_57452_, Player p_57453_, InteractionHand p_57454_, BlockHitResult p_57455_) {
        for (int i = 0; i < dropSize; i++) {
            // 現在持ってるアイテムをコピーする
            ItemStack currentItem = p_57453_.getMainHandItem().copy();
            // 一個だけ
            currentItem.setCount(1);
            // アイテムを地面に生成
            Block.popResource(p_57451_, p_57452_, currentItem);
        }
        return InteractionResult.SUCCESS;
    }
}

File: src/main/kotlin/io/github/takusan23/clickmanaita/block/ClickManaitaBaseBlock.kt
package io.github.takusan23.clickmanaita.block

import net.minecraft.core.BlockPos
import net.minecraft.world.InteractionResult
import net.minecraft.world.entity.player.Player
import net.minecraft.world.level.Level
import net.minecraft.world.level.block.Block
import net.minecraft.world.level.block.state.BlockState
import net.minecraft.world.phys.BlockHitResult

/**
 * ブロックをクリックしたら持ってるアイテムが増える
 *
 * @param settings マテリアルの指定とか
 * @param dropSize 増やす数など
 * */
class ClickManaitaBaseBlock(settings: Properties, private val dropSize: Int = 2) : Block(settings) {

    /** ブロックを右クリックしたら呼ばれる */
    override fun useWithoutItem(blockState: BlockState, level: Level, blockPos: BlockPos, player: Player, blockHitResult: BlockHitResult): InteractionResult {
        return if (level.isClientSide) {
            InteractionResult.CONSUME
        } else {
            repeat(dropSize) {
                // 今持ってるアイテム
                val currentItem = player.mainHandItem.copy()
                currentItem?.count = 1
                // アイテムを落とす
                popResource(level, blockPos, currentItem)
            }
            InteractionResult.SUCCESS
        }
    }

}